marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ZipSplitter.java

```diff
@@ -0,0 +1,363 @@
+            }
+            this.extensionFormats = extensionFormats;
+        }
+
+        Predicate<ZipEntry> getEntryFilter() {
+            return this.entryFilter;
+        }
+
+        void setEntryFilter(Predicate<ZipEntry> entryFilter) {
+            this.entryFilter = entryFilter;
+        }
+
+        Function<String, String> getUriTransformer() {
+            return this.uriTransformer;
+        }
+
+        void setUriTransformer(Function<String, String> uriTransformer) {
+            this.uriTransformer = uriTransformer;
+        }
+
+        protected FormatEntry getNextEntry() throws IOException {
+            ZipEntry candidateEntry;
+
+            while ((candidateEntry = getZipStream().getNextEntry()) != null) {
+                if (getEntryFilter() != null && getEntryFilter().test(candidateEntry) == false) {
+                    continue;
+                }
+
+                String name = candidateEntry.getName();
+                if (getUriTransformer() != null) {
+                    name = getUriTransformer().apply(name);
+                }
+
+                Matcher matcher = extensionRegex.matcher(name);
+                matcher.find();
+                String extension = matcher.group(1);
+                Format format = getExtensionFormats().get(extension);
+
+                if (format == null) {
+                    format = getExtensionFormats().get("");
+                }
+
+                if (format == null || format == Format.UNKNOWN) {
+                    continue;
+                }
+
+                FormatEntry newEntry = new FormatEntry();
+                newEntry.setFormat(format);
+                newEntry.setZipEntry(candidateEntry);
+
+                return newEntry;
+            }
+
+            return null;
+        }
+
+        protected BytesHandle readEntry(FormatEntry entry) throws IOException {
+            byte[] content = new byte[(int) entry.getZipEntry().getSize()];
+            getZipStream().read(content, 0, content.length);
+            return new BytesHandle(content).withFormat(entry.getFormat());
+        }
+    }
+
+    private static class BytesHandleSpliterator extends ZipEntrySpliterator<BytesHandle> {
+
+        private ZipSplitter splitter;
+
+        protected BytesHandleSpliterator(ZipSplitter splitter) {
+            super(Long.MAX_VALUE, Spliterator.NONNULL + Spliterator.IMMUTABLE);
+            this.splitter = splitter;
+        }
+
+        @Override
+        public boolean tryAdvance(Consumer<? super BytesHandle> action) {
+            try {
+                FormatEntry nextEntry = getNextEntry();
+                if (nextEntry == null) {
+                    return false;
+                }
+
+                BytesHandle nextBytesHandle = readEntry(nextEntry);
+                action.accept(nextBytesHandle);
+                splitter.count++;
+
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
The IDEs tend to generate boilerplate code that suppresses exceptions.

Unless we know it is safe to suppress an exception, the best practice is to throw the exception. In this case, if we encounter an IOException, something must be wrong with the zipfile.

Where the exception is a checked exception (like IOException) that's not allowed by the signature (as with the tryAdvance() method), we can rethrow it with something like

    throw new RuntimeException("Could not read ZipEntry", e);

To be clear, the same issue affects the other tryAdvance() method.

Reply:
Replaced "e.printStackTrace();" with "throw new RuntimeException("Could not read ZipEntry", e);"